src/main/java/org/eolang/speco/Speco.java

```diff
@@ -24,129 +24,20 @@
-     */
-    private final Path input;
-
-    /**
-     * Absolute path to the directory with output files.
-     */
-    private final Path output;
-
-    /**
-     * Flag indicating whether the input files is EO-program.
-     */
-    private final boolean eolang;
-
-    /**
-     * Ctor.
-     *
-     * @param inp Path to the directory with input files
-     * @param out Path to the directory with output files
-     * @param eolang Iff the input program is in EO
-     */
-    Speco(final Path inp, final Path out, final boolean eolang) {
-        this.input = inp.toAbsolutePath();
-        this.output = out.toAbsolutePath();
-        this.eolang = eolang;
-    }
-
-    /**
-     * Starts the specialization process.
-     *
-     * @throws IOException In case of errors when working with files or parsing a document
-     */
-    void exec() throws IOException {
-        final Path source;
-        if (this.eolang) {
-            source = Speco.parse(this.input);
-        } else {
-            source = this.input;
-        }
-        for (final Path path : Files.newDirectoryStream(source)) {
-            final String transformed = Speco.applyTrain(
-                new Xsline(
-                    new TrDefault<Shift>().with(
-                        new StClasspath("/org/eolang/parser/wrap-method-calls.xsl")
-                    )
-                ).pass(new XMLDocument(Files.readString(path)))
-            ).toString();
-            final String after;
-            if (this.eolang) {
-                after = new XMIR(transformed).toEO();
-            } else {
-                after = transformed;
-            }
-            Files.createDirectories(this.output);
-            Files.write(this.output.resolve(path.getFileName()), after.getBytes());
-        }
-    }
-
-    /**
-     * Applies train of XSL-transformations.
-     *
-     * @param xml XML
-     * @return XML
-     */
-    private static XML applyTrain(final XML xml) {
-        final Train<Shift> train = new TrDefault<Shift>()
-            .with(new StClasspath("/org/eolang/speco/1-1-coping.xsl"))
-            .with(new StEndless(new StClasspath("/org/eolang/speco/1-2-specialization.xsl")))
-            .with(new StClasspath("/org/eolang/speco/1-3-extension.xsl"))
-            .with(new StClasspath("/org/eolang/speco/2-1-substitute-applications.xsl"))
-            .with(new StClasspath("/org/eolang/speco/3-1-add-with.xsl"))
-            .with(new StClasspath("/org/eolang/speco/4-1-fence-tuples.xsl"))
-            .with(new StClasspath("/org/eolang/speco/5-1-substitute-fence.xsl"))
-            .with(new StClasspath("/org/eolang/speco/6-1-substitute-dominant.xsl"))
-            .with(new StClasspath("/org/eolang/speco/7-1-substitute-returned.xsl"));
-        return new Xsline(train).pass(xml);
-    }
-
+public interface Speco {
     /**
-     * Takes source codes on EO, converts to xmir and applies the AOI tool.
+     * Applies XSL-transformations to file.
```

Comment:
@advasileva It looks like we have `xml` here, it's not a file.

Reply:
@volodya-lombrozo fixed